pkg/kubelet/volume/nfs/nfs.go

```
@@ -0,0 +1,200 @@
+	return nfsPluginName
+}
+
+func (plugin *nfsPlugin) CanSupport(spec *api.Volume) bool {
+	if spec.VolumeSource.NFS != nil {
+		return true
+	}
+	return false
+}
+
+func (plugin *nfsPlugin) NewBuilder(spec *api.Volume, podUID types.UID) (volume.Builder, error) {
+	return plugin.newBuilderInternal(spec, podUID, NewNFSMounter())
+}
+
+func (plugin *nfsPlugin) newBuilderInternal(spec *api.Volume, podUID types.UID, mounter NFSMountInterface) (volume.Builder, error) {
+	return &nfs{
+		volName:      spec.Name,
+		server:       spec.VolumeSource.NFS.Server,
+		exportPath:   spec.VolumeSource.NFS.SourcePath,
+		mountOptions: spec.VolumeSource.NFS.MountOptions,
+		mounter:      mounter,
+		podUID:       podUID,
+		plugin:       plugin,
+	}, nil
+}
+
+func (plugin *nfsPlugin) NewCleaner(volName string, podUID types.UID) (volume.Cleaner, error) {
+	return plugin.newCleanerInternal(volName, podUID, NewNFSMounter())
+}
+
+func (plugin *nfsPlugin) newCleanerInternal(volName string, podUID types.UID, mounter NFSMountInterface) (volume.Cleaner, error) {
+	return &nfs{
+		volName:      volName,
+		server:       "",
+		exportPath:   "",
+		mountOptions: "",
+		mounter:      mounter,
+		podUID:       podUID,
+		plugin:       plugin,
+	}, nil
+}
+
+// NFS volumes represent a bare host file or directory mount of an NFS export.
+type nfs struct {
+	volName      string
+	podUID       types.UID
+	server       string
+	exportPath   string
+	mountOptions string
+	mounter      NFSMountInterface
+	plugin       *nfsPlugin
+}
+
+func (nfs *nfs) SetUp() error {
+	mountpoint, err := mount.IsMountPoint(nfs.GetPath())
+	glog.V(4).Infof("NFS mount set up: %s %v %v", nfs.GetPath(), mountpoint, err)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	if mountpoint {
+		return nil
+	}
+	exportDir := nfs.exportPath
+	mountDir := nfs.GetPath()
+	exists, err := exists(mountDir)
+	if err != nil {
+		glog.Errorf("Mount directory is invalid: %v, %v", mountDir, err)
```

I think we want to just return the error in this case.